core/src/main/scala/kafka/server/DelayedFetch.scala

```diff
@@ -169,22 +215,42 @@ class DelayedFetch(
       tp -> status.fetchInfo
     }
 
+    val (inklessFetchInfos, classicFetchInfos) = fetchInfos.partition { case (k, _) => isInklessTopic(k.topic()) }
+    val inklessParams = replicaManager.fetchParamsWithNewMaxBytes(params, inklessFetchInfos.size.toFloat / fetchInfos.size.toFloat)
+    val classicParams = replicaManager.fetchParamsWithNewMaxBytes(params, classicFetchInfos.size.toFloat / fetchInfos.size.toFloat)
+
+    val inklessFetchResponseFuture = replicaManager.fetchInklessMessages(inklessParams, inklessFetchInfos)
+
     val logReadResults = replicaManager.readFromLog(
-      params,
-      fetchInfos,
+      classicParams,
+      classicFetchInfos,
       quota,
       readFromPurgatory = true
     )
 
-    val fetchPartitionData = logReadResults.map { case (tp, result) =>
-      val isReassignmentFetch = params.isFromFollower &&
-        replicaManager.isAddingReplica(tp.topicPartition, params.replicaId)
+    val classicFetchPartitionData = logReadResults.map { case (tp, result) =>
+      val isReassignmentFetch = classicParams.isFromFollower &&
+        replicaManager.isAddingReplica(tp.topicPartition, classicParams.replicaId)
 
       tp -> result.toFetchPartitionData(isReassignmentFetch)
     }
 
-    responseCallback(fetchPartitionData)
+    inklessFetchResponseFuture.whenComplete{ case (inklessFetchPartitionData, _) =>
+      responseCallback(classicFetchPartitionData ++ inklessFetchPartitionData)
+    }
+
   }
+
+  private def safeLongToInt(longValue: Long): Int = {
```

Comment:
Don't need to change here, but when we contribute to Kafka, it'll probably be in `Utils`